crates/storage/provider/src/providers/database/metrics.rs

```diff
@@ -0,0 +1,66 @@
+use metrics::Histogram;
+use reth_metrics::Metrics;
+use std::time::{Duration, Instant};
+
+#[derive(Debug)]
+pub(crate) struct DurationsRecorder {
+    pub(crate) actions: Vec<(Action, Duration)>,
+    latest: Instant,
+}
+
+impl Default for DurationsRecorder {
+    fn default() -> Self {
+        Self { actions: Vec::new(), latest: Instant::now() }
+    }
+}
+
+impl DurationsRecorder {
+    /// Saves the provided duration for future logging and instantly reports as a metric with
+    /// `action` label.
+    pub(crate) fn record_duration(&mut self, action: Action, duration: Duration) {
+        self.actions.push((action, duration));
+        Metrics::new_with_labels(&[("action", format!("{action:?}"))]).duration.record(duration);
+        self.latest = Instant::now();
```

Comment:
if we only record_relative then isn't required if we record the last duration instead, or am I missing something?
because this would be the same as deducting the last recorded elapsed duration from the next elapsed: `elpased() - last_elapased`

see my original review

Reply:
what does last duration mean? Doesn't it still require doing `Instant::now()` and comparing against some instant in the past?

`elapsed() - last_elapsed` also requires to do at least one `Instant::now()`, no?

Reply:
the way we currently use this is like

```rust

// 0. create instant
let mut r = Recorder {now: Instant::now(), latest: None};

// 1. records duration from [0..1] 
r.record_relative(MakeCanonicalAction::FindCanonicalHeader);
// r.latest is now updated to `now.elapsed()`

....

// 2. record duration from [1..2] by `now.elapsed() - latest`


```

Reply:
ah yeah, that's better